src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -296,7 +296,7 @@ internal MatchInfo Clone()
     /// A cmdlet to search through strings and files for particular patterns.
     /// </summary>
     [Cmdlet(VerbsCommon.Select, "String", DefaultParameterSetName = "File", HelpUri = "https://go.microsoft.com/fwlink/?LinkID=113388")]
-    [OutputType(typeof(MatchInfo), typeof(bool))]
+    [OutputType(typeof(MatchInfo), typeof(bool), typeof(string))]
```

Comment:
If you look at `TestConnectionCommand` for a moment, it has a great example of how to work with OutputType -- you can actually specify the output type(s) based on the parameter set name.

I would redefine it to have the MatchInfo/bool type in Object & File sets, and then give it string output type for the ObjectRaw / FileRaw sets. :)

Reply:
Oh, I did not know that was even possible! Neat! Will look at that, thanks.

Reply:
So looks like this is the only remaining thing for this PR and then we're merging it.  or let me know if this is not blocking.

Reply:
I'm going to work on this tonight. I have no idea if this would be blocking or not. That's not up to me either I think 🙂

Reply:
@vexx32, I've pushed changes, could you have a look please? The problem I ran into is, that because `-Quiet` is not in it's own paramter set, I don't think we can get `OutputType` to be more specific than this. What do you think?

Reply:
I've also gone ahead and refactored the parameter set names to string constants, just like in TestConnectionCommand.cs. It makes the code more readable and easier to refactor later.